File: manhunt-base/src/main/java/io/github/ytg1234/manhunt/base/Manhunt.java
package io.github.ytg1234.manhunt.base;

import io.github.ytg1234.manhunt.base.config.ManhuntConfig;
import io.github.ytg1234.manhunt.base.init.ManhuntEventRegistration;
import me.sargunvohra.mcmods.autoconfig1u.AutoConfig;
import me.sargunvohra.mcmods.autoconfig1u.serializer.JanksonConfigSerializer;
import net.fabricmc.api.ModInitializer;

import static io.github.ytg1234.manhunt.base.ManhuntUtils.LOGGER;

public class Manhunt implements ModInitializer {

    /**
     * Initializes the mod.
     */
    @Override
    public void onInitialize() {
        LOGGER.info("Initializing");
        AutoConfig.register(ManhuntConfig.class, JanksonConfigSerializer::new);
        ManhuntUtils.CONFIG = AutoConfig.getConfigHolder(ManhuntConfig.class).getConfig();

        ManhuntEventRegistration.registerCommonEvents();
    }
}

File: manhunt-base/src/main/kotlin/io/github/ytg1234/manhunt/base/Manhunt.kt
package io.github.ytg1234.manhunt.base

import io.github.ytg1234.manhunt.base.init.ManhuntEventRegistration
import org.quiltmc.loader.api.ModContainer
import org.quiltmc.qsl.base.api.entrypoint.ModInitializer

/**
 * Initializes the mod.
 *
 * @author YTG1234
 */
object Manhunt : ModInitializer {
    override fun onInitialize(mod: ModContainer?) {
        LOGGER.info("Initializing")
        ManhuntEventRegistration.registerCommonEvents()
    }
}
